workspace/filtering.md

```diff
@@ -0,0 +1,72 @@
+# Workspace filtering
+
+When running commands on a workspace, it is possible to limit them to a subset of packages.
+pnpm supports package selectors that can be used together with its `recursive` commands:
+
+```sh
+pnpm recursive <cmd> [arg..] -- [selector...]
+# or
+pnpm recursive <cmd> [arg..] --filter <selector-1> --filter <selector-2>...
+```
+
+When packages in the workspace are filtered, every package is taken that matches at least one of
+the selectors.
+
+## Selecting by path
+
+Selectors that start with `./` are selectors by filesystem path. Selectors by path don't support globs.
+
+### Examples
+
+Select all packages in the `packages` folder:
+
+```sh
+pnpm recursive install -- ./packages
+```
+
+## Selecting by package name
+
+Selectors that don't start with `./` are selectors by package name.
+
+### Selecting only the specified packages
+
+To select an exact package, just specify its name. For insance,
+the following command will be applied on `foo`, `bar` and `@pnpm/config`:
+
+```
+pnpm recursive install -- foo bar @pnpm/config
+```
+
+### Selecting a package with dependencies
+
+To select a package and its dependencies, prefix the package name with two dots: `..<pkg name>`.
+For instance, the next command will run installation in all dependencies of `foo` and in `foo`:
+
+```
+pnpm recursive install -- ..foo
```

Comment:
@vjpr thinks it should be the other way around. This is dependents and `foo..` is dependencies. Who else thinks this should be vice versa?